### PR TITLE
fix: Unable to run saveNativeModel for VWRegressionModel #1364

### PR DESCRIPTION
**Pull request by memoryz**

### CODE REVIEW

File: core/src/main/scala/com/microsoft/azure/synapse/ml/io/binary/BinaryFileFormat.scala
Comment (memoryz) [inline core/src/main/scala/com/microsoft/azure/synapse/ml/io/binary/BinaryFileFormat.scala:231]: The bunch of `.getParent` calls are unnecessary. Spark's output committer will move the files to the main output directory. Having these calls doesn't work in cloud environment.

File: core/src/main/scala/org/apache/spark/ml/source/image/PatchedImageFileFormat.scala
Comment (memoryz) [inline core/src/main/scala/org/apache/spark/ml/source/image/PatchedImageFileFormat.scala:78]: Not related to the bug fix, but changed the function into functional style and add tailrec annotation.

**Review by eisber — APPROVED** [review]
VW changes look good to me. Assume this ran through a unit test?

**Comment by memoryz** [comment]
> VW changes look good to me. Assume this ran through a unit test?

yes. The function is covered by unit test and gets run in CI build.

**Comment by memoryz** [comment]
@eisber Markus, the merge is blocked because it requires Mark's approval as well and he's out of office. Can you force merge?

**Review by eisber — APPROVED** [review]

**Comment by eisber** [comment]
unfortunately I cannot. maybe @serena-ruan  can help?

**Review by mhamilton723 — APPROVED** [review]

**Pull request merged**
